Library/ViewModels/LiveStreamContainerViewModel.swift

```diff
@@ -124,202 +124,342 @@ LiveStreamContainerViewModelInputs, LiveStreamContainerViewModelOutputs {
+      .map(LiveStreamType.hlsStream)
+
+    let liveHlsUrl = hlsUrlEvent.values()
+      .map(LiveStreamType.hlsStream)
+
+    let hlsStreamUrl = Signal.merge(liveHlsUrl, replayHlsUrl)
+
+    let openTokSessionConfig = updatedEventFetch.values().map { $0.openTok }
+      .skipNil()
+      .map {
+        LiveStreamType.openTok(
+          sessionConfig: OpenTokSessionConfig(
+            apiKey: $0.appId, sessionId: $0.sessionId, token: $0.token
+          )
         )
     }
 
-    let hideWhenReplay = Signal.merge(
-      self.viewDidLoadProperty.signal.mapConst(true),
-      event.map { !$0.liveNow }
-    ).skipRepeats()
+    let liveStreamType = Signal.merge(
+      Signal.combineLatest(hlsStreamUrl, useHlsStream.filter(isTrue)).map(first),
+      Signal.combineLatest(openTokSessionConfig, useHlsStream.filter(isFalse)).map(first)
+      )
+      .skipRepeats()
 
-    let hideWhenLive = Signal.merge(
+    let observedGreenRoomOffOrInReplay = Signal.merge(
+      greenRoomOffStatus.filter(isTrue),
+      didLiveStreamEndedNormally.filter(isTrue)
+      )
+      .ignoreValues()
+
+    self.createVideoViewController = Signal.combineLatest(
+      liveStreamType,
+      observedGreenRoomOffOrInReplay
+      )
+      .map(first)
+
+    self.disableIdleTimer = Signal.merge(
       self.viewDidLoadProperty.signal.mapConst(true),
-      event.map { $0.liveNow }
-    ).skipRepeats()
+      self.viewDidDisappearProperty.signal.mapConst(false)
+    )
 
-    self.navBarTitleViewHidden = Signal.merge(
-      project.mapConst(true),
-      liveStreamControllerState.map { state in
-        switch state {
-        case .live(playbackState: .playing, _):   return false
-        case .replay(playbackState: .playing, _): return false
-        default:                                  return true
-        }
+    self.removeVideoViewController = self.createVideoViewController.take(first: 1)
+      .sample(on: greenRoomOffStatus.filter(isFalse).ignoreValues())
+      .ignoreValues()
+
+    let greenRoomState = greenRoomOffStatus
+      .filter(isFalse)
+
+    let replayState = didLiveStreamEndedNormally
+      .takePairWhen(self.videoPlaybackStateChangedProperty.signal.skipNil())
+      .filter { didEndNormally, playbackState in didEndNormally && !playbackState.isError }
+
+    let liveState = updatedEventFetch.values()
+      .takePairWhen(self.videoPlaybackStateChangedProperty.signal.skipNil())
+      .filter { event, playbackState in
+        event.liveNow && !playbackState.isError
       }
-    ).skipRepeats()
 
-    self.navBarLiveDotImageViewHidden = hideWhenReplay
-    self.creatorAvatarLiveDotImageViewHidden = hideWhenReplay
-    self.numberWatchingBadgeViewHidden = hideWhenReplay
-    self.availableForLabelHidden = Signal.combineLatest(nonStarter, hideWhenLive).map { $0 || $1 }
+    let nonStarterState = updatedEventFetch.values()
+      .map { event in isNonStarter(event: event) }
+      .filter(isTrue)
+
+    let signInAnonymouslyEvent = Signal.merge(
```

Comment:
So as we spoke about (moved here for discussion and removed the fixme). The code from line 336 - 376 does the following:

- Signs into Firebase anonymously if `currentUser` is nil. This is to still update the number of people watching value anonymously.
- Signs in with our custom token if it's available.
- Observes the values from the producer to keep it alive as we don't do anything else with it.

Our discussion was how to test this and whether or not we want some kind of output for that.